### PR TITLE
Feat: RTL support

### LINKED ISSUE #367: RTL support
im facing the gesture issue in RTL mode, we need to swipe in opposite direction
im using this with react native tab view and react navigation

### PR DESCRIPTION
**Pull request by mikeqcp**
# Summary

Adds `layoutDirection` property. It switches slides direction between `LTR` or `RTL`.

Closes #367

## Test Plan

https://user-images.githubusercontent.com/746000/126314550-78728761-c2f4-469a-b7f0-4be53793fa1d.mov

### What's required for testing (prerequisites)?

Run `Basic Example` with `layoutDirection` set to `rtl`.
Optionally set `layoutDirection` to `locale` if your device uses `RTL` locale.

### What are the steps to reproduce (after prerequisites)?

Navigate views with swipes or with `Next/Previous/Start/End` buttons - see direction of slides is inverted.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [x] I updated the typed files (TS and Flow)

### CODE REVIEW

File: ios/ReactNativePageView.m [hunk 1]
```diff
@@ -217,7 +219,8 @@ - (void)goTo:(NSInteger)index animated:(BOOL)animated {
         return;
     }
     
-    UIPageViewControllerNavigationDirection direction = (index > self.currentIndex) ? UIPageViewControllerNavigationDirectionForward : UIPageViewControllerNavigationDirectionReverse;
+    BOOL animateForward = (index > self.currentIndex && [self isLtrLayout]) || (index < self.currentIndex && ![self isLtrLayout]);
```
Comment (troZee) [inline ios/ReactNativePageView.m:222]:
```suggestion
    BOOL isForward = (index > self.currentIndex && self.isLtrLayout) || (index < self.currentIndex && !self isLtrLayout);
```

**Review by troZee — CHANGES_REQUESTED** [review]

File: ios/ReactNativePageView.m (hunk 1 above)
Comment (mikeqcp) [reply ios/ReactNativePageView.m]: Done 👍🏼

**Review by troZee — APPROVED** [review]

**Comment by aleppos** [comment]
image

**Comment by mikeqcp** [comment]
@aleppos can you explain how are you getting this error? The only way I'm able to reproduce this is by setting `layoutDirection={null}`, but it is not a valid prop value. Only accepted values are `'ltr' / 'rtl' / 'locale'`

**Pull request merged**